### PR TITLE
Feature filterdesigner

### PR DESCRIPTION
**Pull request by benNthen**
Expanded on GUI functions of filter-panel. Sorting categories now works - items for a category selected by user are listed. Bug encountered is that the Categories List still remains intact resulting in Items of the selected category being generated on-top of the list.

### CODE REVIEW

**Comment by KadinHoneyfield** [comment]
Hey Ben.
Check your commits. I can't see any evidence of the added features in the files that have been changed.

**Comment by benNthen** [comment]
Hey Kadin! ok I'll check it out soon

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

File: Assets/scripts/LoadCatalog.cs [hunk 1]
```diff
@@ -275,6 +273,52 @@ private void ChangeContentToCategory(string category)
...
+        {
+            if (itemInList.GetCategories().Contains("Google"))
+            {
+                showListing(itemInList);
+                content.transform.Find($"Listing: {itemInList.GetItemID()} {itemInList.GetName()}").gameObject.SetActive(true);
+            }
+        }
+    }
+
+
```
Comment (KadinHoneyfield) [inline Assets/scripts/LoadCatalog.cs]: These methods are not needed as the ChangeContentToCategory method takes in a string parameter (such as "Google", "Office", or "Living Room"). For example: calling categoryGoogle() will do the exact same thing as ChangeContentToCategory("Google")

Try and make it so when you click on a filter it will call the ChangeContentToCategory() method with the category string in the parameters.

File: Assets/scripts/LoadCatalog.cs [hunk 2]
```diff
@@ -263,8 +263,6 @@ private void ChangeContentToCategory(string category)
     {
         GameObject content = GameObject.Find("Content");
 
-        hideListings();
-
```
Comment (KadinHoneyfield) [inline Assets/scripts/LoadCatalog.cs]: Why is this removed? This method hides all listings currently visible which should be done before you start showing other listings. This is probably why when you select a filter the old results are still visible.

**Review by KadinHoneyfield — COMMENTED** [review]
Think I've found why the listings are being overlapped.
Overall I don't see how this is filtering for "designer".
We need to create a "designer" string in the item class and add a method similar to ChangeContentToCategory for searching the new designer field in the item class. This would be similar for filtering by brand.
Any questions let me know :)

File: Assets/scripts/LoadCatalog.cs (hunk 1 above)
Comment (benNthen) [reply Assets/scripts/LoadCatalog.cs]: Alright, I'll trim this down.

File: Assets/scripts/LoadCatalog.cs (hunk 2 above)
Comment (benNthen) [reply Assets/scripts/LoadCatalog.cs]: My bad. Probably did a cut-paste instead of copy-paste in the process of testing out the methods last night. I just returned it now and its working fine.

File: Assets/scripts/LoadCatalog.cs (hunk 1 above)
Comment (benNthen) [reply Assets/scripts/LoadCatalog.cs]: Alright I'll trim them all down. Thanks for the feedback. Will keep you guys updated :)

**Comment by benNthen** [comment]
@KadinHoneyfield yes the resetListing() method where I used resetListing() and Start() is to reset the catalogue view. Just an added option for the user, if they want to revert the changes made on the listing upon filtering(eg.  Sort to 'Any' instead). But I can remove this function if its actually not needed.

**Comment by benNthen** [comment]
@MillanUka Ok I'll try it.

File: Assets/scripts/LoadCatalog.cs [hunk 3]
```diff
@@ -259,6 +296,56 @@ private void GenerateCategories()
         Debug.Log($"Categories loaded: {result}");
     }
 
+    private void GenerateBrands()
+    {
+        foreach (Item item in loadedItems)
+        {
+            List<string> itemBrands = item.GetBrand();
+
+            foreach (string brand in itemBrands)
```
Comment (MinhPhu0304) [inline Assets/scripts/LoadCatalog.cs]: Please move the second for each loop into another function to make it easier  to follow

**Review by MinhPhu0304 — CHANGES_REQUESTED** [review]

File: Assets/scripts/FilterOpen.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    public void showAny()
+    {
+        LoadCatalog scriptToAccess = GameObject.Find("Viewport").GetComponent<LoadCatalog>();
+
+        scriptToAccess.resetListing();
+    }
+
+    public void showCategories(string categoryName)
+    {
+        LoadCatalog scriptToAccess = GameObject.Find("Viewport").GetComponent<LoadCatalog>();
```
Comment (MinhPhu0304) [inline Assets/scripts/FilterOpen.cs]: I think  the variable name is not clear for me. Something like CatalogScene or something similar would be more clear.

File: Assets/scripts/LoadCatalog.cs [hunk 5]
```diff
@@ -275,6 +372,174 @@ private void ChangeContentToCategory(string category)
...
+        }
+    }
+
+    private void ChangeContentToDesigner(string designer)
+    {
+        GameObject content = GameObject.Find("Content");
+
+        hideListings();
+
+        foreach (Item itemInList in loadedItems)
```
Comment (MinhPhu0304) [inline Assets/scripts/LoadCatalog.cs]: I think the variable should only be item not itemInList because it's kinda obvious when read the code as for each item in loadedItems.

**Review by MinhPhu0304 — CHANGES_REQUESTED** [review]

File: Assets/scripts/FilterOpen.cs (hunk 4 above)
Comment (benNthen) [reply Assets/scripts/FilterOpen.cs]: Ok I'll rename them later

File: Assets/scripts/LoadCatalog.cs (hunk 5 above)
Comment (benNthen) [reply Assets/scripts/LoadCatalog.cs]: Ok got it. Will push another commit sometime later today with the variables renamed :-)
